featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -658,19 +653,22 @@ def _max_hlevel(self, f):
 def check_stacking(primitive, input_types):
     """checks if features in input_types can be used with supplied primitive
        using the stacking rules"""
+    assert primitive.__class__ != str.__class__
 
     if primitive.stack_on_self is False:
         for f in input_types:
-            if isinstance(f.primitive, primitive):
+            if isinstance(f.primitive, primitive.__class__):
                 return False
 
     if primitive.stack_on_exclude is not None:
         for f in input_types:
             if isinstance(f.primitive, tuple(primitive.stack_on_exclude)):
                 return False
 
+    # R TODO: handle this
     for f in input_types:
-        if f.primitive.expanding:
+        feats = f.get_dependencies(deep=True) + [f]
+        if any([feat.primitive.number_output_features > 1 for feat in feats]):
```

Comment:
i don't think you need to check the deep dependencies. if the input feature doesn't have `number_output_features > 1` it shouldn't matter what it's dependents have.